### DISCUSSION TITLE
How to get old docker minecraft server to match old client?

### DISCUSSION THREAD

**Discussion by TonySerpofski** [itzg/docker-minecraft-server#2868] [General]
Im trying to get a minecraft server set up for my kids. Which was usually a simple task on Docker but its thrown a wobbler and made me question my understanding of images - Maybe you can help?

Im trying to run an old server with a docker tag to match 1.20.4 client. It threw an error:

```
Error: LinkageError occurred while loading main class net.minecraft.bundler.Main
`java.lang.UnsupportedClassVersionError: net/minecraft/bundler/Main has been compiled by a more recent version of the Java Runtime (class file version 65.0), this version of the Java Runtime only recognizes class file versions up to 61.0`
```

I thought that docker was meant to have everything necessary in the container to run? Is this some security mechanism to ensure people only run up-to-date clients? So i have to roll my own container to do this task?

How do i match this client with a [server](https://hub.docker.com/r/itzg/minecraft-server/tags)?

Thanks

**Comment by itzg** [comment]
Re-pulling the image or using java21 image tag will resolve that

https://docker-minecraft-server.readthedocs.io/en/latest/versions/java/

If you need to actually go to an older Minecraft version, just changing VERSION env var should be enough 

https://docker-minecraft-server.readthedocs.io/en/latest/versions/minecraft/

**Comment by TonySerpofski** [comment]
this worked - thanks to your help!

services:
  minecraft:
    image: itzg/minecraft-server:stable
    restart: always
    container_name: minecraft
    ports:
      - "25565:25565"
    environment:
      EULA: "TRUE"
      GAMEMODE: creative
      DIFFICULTY: easy
      LEVEL_NAME: newland
      SERVER_NAME: mc
      VERSION: 1.20.4
      ONLINE_MODE: FALSE
